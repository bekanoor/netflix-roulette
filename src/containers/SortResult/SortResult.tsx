import React from 'react'

interface IProps {
  movieLength?: number
  onClick: (value: string) => void
}

const SortResult = (props: IProps) => {
  const { movieLength, onClick } = props

  const switchColor = (event: React.MouseEvent<HTMLButtonElement>): void => {
    const eventTarget = event.currentTarget as HTMLElement
    if (eventTarget.classList.contains('sort-result--active')) {
      return
    }

    const buttons = document.querySelectorAll('.sort-result__item')
    const firstBtn = buttons[0]
    const secondBtn = buttons[1]

    const condition = firstBtn.classList.contains('sort-result--active')

    if (condition) {
      firstBtn.classList.remove('sort-result--active')
      secondBtn.classList.add('sort-result--active')
    } else {
      secondBtn.classList.remove('sort-result--active')
      firstBtn.classList.add('sort-result--active')
    }
  }

  const handlerChange = (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ): void => {
    const eventTarget = event.target as HTMLElement
    switchColor(event)
    onClick(eventTarget.innerText)
  }

  return (
    <div className='sort-result' data-testid='sort-result-test'>
      <p className='sort-result__text'>{movieLength} movies found</p>
      <div>
        <ul className='sort-result__list'>
          <li className='sort-result__text'>Sort by</li>
          <li>
            <button onClick={handlerChange} className='sort-result__item'>
              release data
            </button>
          </li>
          <li>
            <button
              onClick={handlerChange}
              className='sort-result__item sort-result--active'
            >
              rating
            </button>
          </li>
        </ul>
      </div>
    </div>
  )
}

export { SortResult }
