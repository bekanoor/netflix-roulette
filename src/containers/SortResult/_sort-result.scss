$primary-red: #F65261;

.sort-result {
  padding: 15px 50px;
  height: 50px;
  background-color: #2c2c2c;
  display: flex;
  justify-content: space-between;
  &__text {
    color: white;
  }
  &__list {
    width: 300px;
    display: flex;
    justify-content: space-between;
  }
  &__item {
    color: white;
    background: none;
    font-size: 20px;
  }
  &--active {
    color: $primary-red;
  }
}