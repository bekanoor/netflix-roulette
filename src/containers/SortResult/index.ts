import { SortResult } from './SortResult'

export { SortResult }
