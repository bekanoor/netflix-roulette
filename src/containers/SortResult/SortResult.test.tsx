import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { SortResult } from './SortResult'

describe('test search result component', () => {
  test('should render search result correctly', () => {
    const {container} = render(<SortResult onClick={()=>{}}/>)

    expect(container).toMatchSnapshot()
  })

  test('dom elements really exist', () => {
    render(<SortResult onClick={()=>{}}/>)

    expect(screen.getByTestId('sort-result-test')).toBeInTheDocument
  }) 
})