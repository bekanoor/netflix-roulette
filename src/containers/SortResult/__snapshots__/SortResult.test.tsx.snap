// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test search result component should render search result correctly 1`] = `
<div>
  <div
    class="sort-result"
    data-testid="sort-result-test"
  >
    <p
      class="sort-result__text"
    >
       movies found
    </p>
    <div>
      <ul
        class="sort-result__list"
      >
        <li
          class="sort-result__text"
        >
          Sort by
        </li>
        <li>
          <button
            class="sort-result__item"
          >
            release data
          </button>
        </li>
        <li>
          <button
            class="sort-result__item sort-result--active"
          >
            rating
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
